src/libraries/System.Private.Xml/src/System/Xml/Schema/DataTypeImplementation.cs

```diff
@@ -1143,6 +1141,15 @@ internal override RestrictionFlags ValidRestrictionFlags
 
         Error:
             return exception;
+
+            [UnconditionalSuppressMessage("AotAnalysis", "IL3050:AotUnfriendlyApi",
+                Justification = "Array type is always present as it is passed in as a parameter.")]
+            Array ToArray(ArrayList values, Type arrayType)
+            {
+                Array array = values.ToArray(arrayType.GetElementType()!);
+                Debug.Assert(array.GetType() == _itemType.ValueType.MakeArrayType());
```

Comment:
```suggestion
            static Array ToArray(ArrayList values, Type arrayType)
            {
                Array array = values.ToArray(arrayType.GetElementType()!);
```

Reply:
I have not realized that moving the assert into the method will make the method non-static.